File: projects/09-payment-webhook-handler/candidate/app/repositories/order_repository.py
from app.models.schemas import Order


class OrderRepository:
    """In-memory store of orders. In production this would be backed by a
    database table owned by the orders service.
    """

    def __init__(self):
        self._orders: dict[int, Order] = {
            101: Order(id=101, status="pending", amount_paid_cents=0),
            102: Order(id=102, status="pending", amount_paid_cents=0),
            103: Order(id=103, status="pending", amount_paid_cents=0),
        }

    def get(self, order_id: int) -> Order | None:
        return self._orders.get(order_id)

    def save(self, order: Order) -> None:
        self._orders[order.id] = order

File: projects/17-order-notification-service/candidate/src/repositories/orderRepository.ts
import { Order } from '../types';

/**
 * In-memory store of orders, keyed by order id. In production this would be
 * backed by a real database table; for this exercise a Map is enough.
 */
export class OrderRepository {
  private readonly ordersById: Map<string, Order> = new Map();

  getById(id: string): Order | undefined {
    return this.ordersById.get(id);
  }

  list(): Order[] {
    return Array.from(this.ordersById.values());
  }

  save(order: Order): Order {
    this.ordersById.set(order.id, order);
    return order;
  }

  /** Test/dev helper: not used by request handlers. */
  clear(): void {
    this.ordersById.clear();
  }
}

/** Default repository instance used by the app. Starts empty — orders are
 * created through the API, there is no seed data for this service. */
export const orderRepository = new OrderRepository();
